homeassistant/components/sensor/geo_rss_events.py

```diff
@@ -0,0 +1,325 @@
+and grouped by category.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.geo_rss_events/
+"""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (STATE_UNKNOWN, CONF_SCAN_INTERVAL,
+                                 CONF_UNIT_OF_MEASUREMENT, CONF_NAME)
+from homeassistant.helpers.entity import Entity, generate_entity_id
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['feedparser==5.2.1', 'haversine==0.4.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CATEGORIES = 'categories'
+CONF_RADIUS = 'radius'
+CONF_URL = 'url'
+
+DEFAULT_ICON = 'mdi:alert'
+DEFAULT_NAME = "Event Service"
+DEFAULT_RADIUS_IN_KM = 20.0
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=5)
+DEFAULT_UNIT_OF_MEASUREMENT = 'Events'
+
+DOMAIN = 'geo_rss_events'
+ENTITY_ID_FORMAT = 'sensor.' + DOMAIN + '_{}'
+# Minimum time between updates from the source.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS_IN_KM): vol.Coerce(float),
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Optional(CONF_CATEGORIES, default=[]): vol.All(cv.ensure_list,
+                                                       [cv.string]),
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT,
+                 default=DEFAULT_UNIT_OF_MEASUREMENT): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL,
+                 default=DEFAULT_SCAN_INTERVAL): cv.time_period
+})
+
+
+def setup_platform(hass, config, add_devices,
+                   discovery_info=None):  # pragma: no cover
+    """Set up the GeoRSS component."""
+    # Grab location from config
+    home_latitude = hass.config.latitude
+    home_longitude = hass.config.longitude
+    url = config.get(CONF_URL)
+    radius_in_km = config.get(CONF_RADIUS)
+    name = config.get(CONF_NAME)
+    categories = config.get(CONF_CATEGORIES)
+    unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
+
+    _LOGGER.debug("latitude=%s, longitude=%s, url=%s, radius=%s",
+                  home_latitude, home_longitude, url, radius_in_km)
+
+    # Initialise update service.
+    data = GeoRssServiceData(hass, home_latitude, home_longitude, url,
+                             radius_in_km)
+
+    # Create all sensors based on categories.
+    devices = []
+    if not categories:
+        device = GeoRssServiceSensor(hass, None, data, name,
+                                     unit_of_measurement)
+        devices.append(device)
+    else:
+        for category in categories:
+            device = GeoRssServiceSensor(hass, category, data, name,
+                                         unit_of_measurement)
+            devices.append(device)
+    add_devices(devices)
```

Comment:
I have made the corresponding change. However, the current construct of sensor and data classes appears to require the very first update to be made outside the async routine; hence I call update on the data object before instantiating the devices.
If this update did not happen, then each sensor would call update on the shared data object independently, but most or none would wait for the result, i.e. the first sensor update yields in empty states for most or all sensors. I have seen at least the 'glances' component using the same approach.